src/link/filelink.cpp

```diff
@@ -24,7 +24,14 @@ FileLink::FileLink(QObject* parent)
 
 void FileLink::_writeData(const QByteArray& data)
 {
+    // Check if we have already opened the file
+    if(!_file.isOpen()) {
+        qCDebug(PING_PROTOCOL_FILELINK) << "File will be opened.";
+        _file.open(QIODevice::ReadWrite);
```

Comment:
personally, I would check the return of this line instead of checking isOpen() again. Matter of preference.

Reply:
Done.